src/control/lib/hardware/sysfs/provider.go

```diff
@@ -251,6 +292,53 @@ func (s *Provider) addVirtualNetDevices(topo *hardware.Topology) error {
 	return nil
 }
 
+func (s *Provider) addNetvscDevices(topo *hardware.Topology) error {
```

Comment:
A lot of the internals of this function seems very similar to addVirtualNetDevices, I think we should extract some 
common code into helper functions to reduce duplication e.g. fetchDevices(path) and addDevices(virtualDevices).

Reply:
- [ ] Try to factorize code of` addNetvscDevices()` and `addVirtualNetDevices()` function